com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/validation/UniquenessValidValidationHelper.java

```diff
@@ -58,4 +59,34 @@ public UniquenessValidValidationHelper(final INameFunction nameFunction) {
     return messages;
   }
 
+  /**
+   * Check whether the possiblyDuplicateObjects (in the given context) contain duplicates.
+   *
+   * @param context
+   *          the context containing the possible duplicates.
+   * @param possiblyDuplicateObjects
+   *          a set of elements of type T, into which one must look for duplicate objects
+   * @param message
+   *          the duplicate error message, in which {0} is the type of T and {1} is the duplicate representation
+   * @return a compound message (a set of simple messages)
+   */
+  public CompoundValidMessage<T> checkDuplicates(final EObject context, final Iterable<T> possiblyDuplicateObjects, final String message) {

```

Comment:
Perhaps we should also do a null check on the context, or add in the javadoc if context must not be null

Reply:
I've added a null check and a null-safe equals call.